tests/suites/cli/active_branch.sh

```diff
@@ -0,0 +1,88 @@
+# Checks juju indicates the correct branch output tabular and json.
+run_indicate_active_branch_no_active() {
+  echo
+
+  file="${TEST_DIR}/indicate-active-branch-no-active.txt"
+
+  ensure "indicate-active-branch-no-active" "${file}"
+
+  export JUJU_DEV_FEATURE_FLAGS=generations
+  juju deploy ubuntu 2>&1
+
+  wait_for "ubuntu" ".applications | keys[0]"
+  STATUS_NO_BRANCHES=$(juju status)
+  STATUS_JQ_NO_BRANCHES=$(juju status --format=json | jq '.branches')
+  check_not_contains "${STATUS_NO_BRANCHES}" "Branch"
+
+  if [ "${STATUS_JQ_NO_BRANCHES}" != null ]; then
+    echo "The status shows branches even though we do not use them yet"
+    exit 1
+  fi
+
+  destroy_model "indicate-active-branch-no-active"
+}
+
+run_indicate_active_branch_active() {
+  echo
+
+  file="${TEST_DIR}/indicate-active-branch-active.txt"
+
+  ensure "indicate-active-branch-active" "${file}"
+
+  export JUJU_DEV_FEATURE_FLAGS=generations
+  juju deploy ubuntu 2>&1
+
+  juju add-branch bla
+  wait_for "ubuntu" ".applications | keys[0]"
+  #  There could be some really small delay, just to make sure.
+  sleep 2
+
+  STATUS_BRANCHES=$(juju status)
+  check_contains "${STATUS_BRANCHES}" "bla\*"
+
+  STATUS_JQ_BRANCHES=$(juju status --format=json | jq '.branches.bla.active')
+
+  if [ "${STATUS_JQ_BRANCHES}" != true ]; then
+    echo "The status does not show active branch"
+    exit 1
+  fi
+
+  juju add-branch testtest
+  wait_for "ubuntu" ".applications | keys[0]"
+  sleep 2
+
+  STATUS_BRANCHES_CHANGED=$(juju status)
+  check_contains "${STATUS_BRANCHES_CHANGED}" "testtest\*"
```

Comment:
In line the calls. The idea being that repeated use should show the same thing. We actually want to call the juju CLI as much as possible here. Inlining is a premature optimisation against what we're trying to test.